src/ucp/api/ucp.h

```diff
@@ -1238,6 +1292,28 @@ ucs_status_t ucp_worker_get_address(ucp_worker_h worker,
                                     ucp_address_t **address_p,
                                     size_t *address_length_p);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Get the address of the worker object.
+ *
+ * This routine returns the address of the worker object.  This address can be
+ * passed to remote instances of the UCP library in order to connect to this
+ * worker. The memory for the address handle is allocated by this function, and
+ * must be released by using @ref ucp_worker_release_address
+ * "ucp_worker_release_address()" routine.
+ *
+ * @param [in]  worker            Worker object whose address to return.
+ * @param [in]  params            User defined @ref ucp_address_params_t configurations
+ *                                for the @ref ucp_address_t "UCP worker address".
+ * @param [out] address_p         A pointer to the worker address.
+ * @param [out] address_length_p  The size in bytes of the address.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_worker_get_address_type(ucp_worker_h worker,
```

Comment:
need better naming, params may contain smth else in the future

Reply:
Something like `ucp_worker_pack_address`?
plz note that we plan to replace original `ucp_worker_get_address` with this func in UCX 2.0.

@yosefe, wdyt about naming? We could use `ucp_worker_pack_address` and then just deprecate `ucp_worker_get_address` in UCX 2.0

Reply:
maybe we can add this to ucp_worker_query?

Reply:
Fetching an address? IMO, it would be quite confusing. We will need to add new `attr->field_mask` for address, plus we will have to specify required address details

Reply:
I like using `ucp_worker_query`

Reply:
> we will have to specify required address details

we have something similar in ucp_ep_create for different types of connection

Reply:
Yes, `ucp_worker_query` is a semantically appropriate call to obtains an address. My point is that we have to provide some parameters for the address we want to obtain. Currently `ucp_worker_query` accepts just attributes, and the only parameter you can provide is `filed_mask`. I'd need to specify what type of devices need to be included to the resulting address. It could be solved by adding `params` parameter to `ucp_worker_query`, but this looks like a bit confusing to me, because fields in `attrs` and `params` need to be synced.

Reply:
What if to make `attrs` as in/out parameter and fill `attr.address_flags` if needed? If `address_flags == 0` or corresponding `filed_mask` is not set but address is requested then return full address

Reply:
yes, we can add address flags.
field_mask will indicate that (address+address_flags) parameters are present in the struct.

Reply:
Considering that we may want to add other parameters for worker address, it looks more confusing to me than using separate call for fetching worker address.